gateway/models/connections.go

```diff
@@ -0,0 +1,272 @@
+		err := tx.Table(tableConnections).
+			Save(c).
+			Error
+		if err != nil {
+			return fmt.Errorf("failed saving connections, reason=%v", err)
+		}
+
+		err = tx.Table(tableEnvVars).Save(EnvVars{OrgID: c.OrgID, ID: c.ID, Envs: c.Envs}).Error
+		if err != nil {
+			return fmt.Errorf("failed updating env vars from connection, reason=%v", err)
+		}
+
+		// remove all rules association
+		err = tx.Exec(`DELETE FROM private.guardrail_rules_connections WHERE org_id = ? AND connection_id = ?`,
+			c.OrgID, c.ID).Error
+		if err != nil {
+			return fmt.Errorf("failed cleaning guard rail rules connections, reason=%v", err)
+		}
+
+		// add new rule associations only if the rule exists
+		var notFoundRules []string
+		for _, ruleName := range rulesAssocList {
+			var result map[string]any
+			err = tx.Raw(`
+			WITH rules AS ( SELECT id FROM private.guardrail_rules WHERE name = ? )
+			INSERT INTO private.guardrail_rules_connections (org_id, connection_id, rule_id)
+			SELECT ?, ?, ( SELECT id FROM rules )
+			WHERE EXISTS ( SELECT id FROM rules )
+			RETURNING *`,
+				ruleName, c.OrgID, c.ID).Scan(&result).Error
+			if err != nil {
+				return fmt.Errorf("failed creating guard rail association, reason=%v", err)
+			}
+			if len(result) == 0 {
+				notFoundRules = append(notFoundRules, ruleName)
+			}
+		}
+		if len(notFoundRules) > 0 {
+			return &ErrNotFoundGuardRailRules{rules: notFoundRules}
+		}
+		return nil
+	})
+}
+
+func DeleteConnection(orgID, name string) error {
+	return DB.Table(tableConnections).
+		Where(`org_id = ? and name = ?`, orgID, name).
+		Delete(&Connection{}).Error
+}
+
+func GetConnectionByNameOrID(orgID, nameOrID string) (*Connection, error) {
+	var conn Connection
+	err := DB.Debug().Model(&Connection{}).Raw(`
+	SELECT
+		c.id, c.org_id, c.name, c.command, c.status, c.type, c.subtype, c.managed_by,
+		c.access_mode_runbooks, c.access_mode_exec, c.access_mode_connect, c.access_schema,
+		c.agent_id, a.name AS agent_name, a.mode AS agent_mode,
+		COALESCE(c._tags, ARRAY[]::TEXT[]) AS _tags,
+		( SELECT envs FROM public.env_vars WHERE id = c.id ) AS envs,
+		COALESCE(dlpc.config, ARRAY[]::TEXT[]) AS redact_types,
+		COALESCE(reviewc.config, ARRAY[]::TEXT[]) AS reviewers,
+		(SELECT array_length(dlpc.config, 1) > 0) AS redact_enabled,
+		COALESCE((
+			SELECT array_agg(r.name) FROM private.guardrail_rules r
+			INNER JOIN private.guardrail_rules_connections rc ON rc.connection_id = c.id AND rc.rule_id = r.id
+			GROUP BY rc.connection_id
+		), ARRAY[]::TEXT[]) AS guardrail_rules,
+		(
+			SELECT json_agg(r.input) FROM private.guardrail_rules r
+			INNER JOIN private.guardrail_rules_connections rc ON rc.connection_id = c.id AND rc.rule_id = r.id
+		) AS guardrail_input_rules,
+		(
+			SELECT json_agg(r.output) FROM private.guardrail_rules r
+			INNER JOIN private.guardrail_rules_connections rc ON rc.connection_id = c.id AND rc.rule_id = r.id
+		) AS guardrail_output_rules
```

Comment:
we don't need to make this complicated join. This complexity is easily replaceable by responding the IDs of the guardrails rules, and the clients request them at their endpoint.

Reply:
- The `guardrail_input_rules` and `guardrail_output_rules` is used internally and  is not exposed to the api.

We already have an endpoint that exposes the guard rail rules, this is the rules associated with a connection.
Are you saying that we should return the `id` of the `private.guardrail_rules_connections` table instead of obtaining their names? The client will need to perform an additional request (e.g.: listing guard rail rules) to obtain the name for this case, are you ok with this? If this is the case IMO we're just shifting complexity to another layer (WebApp / Clients)

Reply:
there's no complexity on this behavior, it's standard REST API plus keeping the concepts transparent and explicit. So yes, let the clients do extra requests